### ISSUE TITLE
Struggling to get sum for each edge coming out of given node in a batch 

### ISSUE THREAD

**Issue by szymonzmyslony** [rusty1s/pytorch_geometric#333]
I am struggling to implement this distance normalisation routine for batched data:

<img width="300" alt="Zrzut ekranu 2019-05-22 o 14 24 33" src="https://user-images.githubusercontent.com/17514265/58188549-b8eae180-7cb0-11e9-8bdb-3cf1ba4ac853.png">

My data loading looks like:

```
for batched_data in test_loader:
    print(gaussianDistanceLayer(batched_data.edge_attr, batched_data.batch))
```

where edge_attr contains distance (x_i - x_j) between 2 nodes. I then try to pass it through gaussianDistanceLayer:

```
class GaussianDistanceLayer(nn.Module):
    def __init__(self):
        super().__init__()
        self.scale = torch.nn.Parameter(torch.randn(1))
        
    def forward(self, batched_ditance, batch):
        result = torch.exp( (-1/2) *batched_ditance.pow(2)/self.scale)
        normalizing_factor = scatter_add(result, index=batch,dim=0)
        return result 
```
However, I don't understand how I should index different nodes in a given graph (which is indexed by batched_data.batch

**Comment by szymonzmyslony** [comment]
Solved. I need to use indexes from edge_indeex

**Comment by rusty1s** [comment]
Exactly :)
